Add Folder Scores page

jxjj opened the pull request:
This adds a page for presenters to see a summary of user participation on questions within a given folder.

### Background
ChimeIn is a great tool for increasing classroom engagement. 

As an instructor, I'd like to see a bird's eye view of which students are participating, and which aren't so that I help those who need it or modify my pedagogical approach. 

Additionally, for graded chimes, if I have questions about how ChimeIn is calculating a particular student's score, I'd like to be able to see the disaggregated participation information (which questions were right/wrong and which had no submission).

The current approach is workable: instructors can use the CSV export feature download a spreadsheet and open it with excel. But, it involves a few steps and remembering which options to choose in the export mode. This PR makes the process a bit easier and faster for the simpler use cases.

### Summary of major changes
- To make it easier to test chimes with many users and responses. this adds factories for questions, responses, and methods for creating participants within chimes 
- Updated the default DB seeder to create a chime with pre-populated activity and users.
- Added api endpoint for getting folder participation info: `'/api/chime/{chime}/folder/{folder}/participation`. This endpoint is only accessible to users who have Presenter permission.
- Added `FolderScoresPage` for viewing folder scores.
- Added button to the `Folder` page for navigating to the `FolderScoresPage`.

### Screenshots

Folder page: Toolbar has new "Scores" button. ("Scores" was chosen since "Participation" seemed too similar to "Participant View".)
![CleanShot 2022-10-27 at 16 51 13@2x](https://user-images.githubusercontent.com/980170/198407605-1e62145a-c1f3-4146-bc08-74b6f5988949.png)

New Folder Scores Page: Full credit for participation view, so all participation shows up as full stars.
![CleanShot 2022-10-27 at 16 51 29@2x](https://user-images.githubusercontent.com/980170/198407623-40e82fd6-c46c-408e-9fb9-51036bf40074.png)

Same data, but set as half credit for incorrect answers. Incorrect responses show up as half-stars.
![CleanShot 2022-10-27 at 16 52 05@2x](https://user-images.githubusercontent.com/980170/198407688-5e281faf-f408-4bf7-9710-5e14365fb3f0.png)

Same data, but chime set as no credit for incorrect answers. Incorrect responses show as very light stars. Questions without response submitted show up as "-".
![CleanShot 2022-10-27 at 16 52 50@2x](https://user-images.githubusercontent.com/980170/198407746-f4f49c42-79d4-4d79-8413-3fd2a637354e.png)

### Other notable changes
- fixed some tests that broke when DB seeding changed. The updated tests are less tightly coupled to DB seed order.
- migrate `router` and `store` to typescript
-  `ltiSetupPreferences` page:
	- tweak design to make it a bit more obvious when a setup choice is selected (the selected item increases in size, easing timing tweaks).
	- some automated linting fixes that happened on save (like moving `<style>` to the end.
- Added helpers to `Chime` model: 
	- `participants()`  method to get all of the Chime's participants (akin to `presenters()` method)
- Added helpers to `Question` model:  
	- `getQuestionType()`
	- `getResponseChoices()`,
	- `isResponseCorrect()`.
	- constants for various question type strings (e.g. `MULTIPLE_CHOICE_TYPE`).
- Added helpers to `Response` model:
	- `getQuestion()`
	- `getQuestionType()` 
- create `Icon` component

### Future Direections

In a future PR, we can add functionality that would allow presenters (or admins) the ability to override scores when needed without creating mock responses in the DB.

cmcfadden commented:
This looks good. Only thing I'm a little hung up on is whether "scores" is the right term? Responses? Stats? You have any other ideas?

jxjj commented:
Changes:
-"Scores" to "Report". Good suggestion.
- When a chime has a lot of participants or questions, it gets difficult to track which score corresponds to which user question. I made the table header and the username columns sticky, and limited the table view container height.
- The black stars were too heavy when looking at a large amount of data. I changed them to gold.
- I changed grayed out star to red x's, which makes for more noticeable contrast against the gold stars.

<img width="1513" alt="CleanShot 2022-10-29 at 09 17 45@2x" src="https://user-images.githubusercontent.com/980170/198836923-26d1be16-18a2-4d92-b911-53b4dce398f7.png">

Also added a spinner while report is generating. It can take awhile to get the data for large classes. We could look at optimizing this a bit.

jxjj commented:
Example chime with fake participants and responses on dev: https://cla-chimein-dev.cla.umn.edu/chime/251/folder/341
